plugins/by-name/lsp-format/default.nix

```diff
@@ -91,28 +90,41 @@ lib.nixvim.plugins.mkNeovimPlugin {
 
   extraConfig = cfg: {
     warnings = lib.nixvim.mkWarnings "plugins.lsp-format" {
-      when = !config.plugins.lsp.enable;
+      when = !(config.plugins.lsp.enable || config.plugins.lspconfig.enable);
       message = ''
-        This plugin requires `plugins.lsp` to be enabled.
+        This plugin requires either `plugins.lsp` or `plugins.lspconfig` to be enabled.
       '';
     };
 
-    plugins.lsp = {
-      onAttach =
-        lib.mkIf (cfg.lspServersToEnable == "all") # Lua
-          ''
-            require("lsp-format").on_attach(client)
-          '';
-
-      servers = lib.optionalAttrs (lib.isList cfg.lspServersToEnable) (
-        lib.genAttrs cfg.lspServersToEnable (serverName: {
-          onAttach.function = # Lua
-            ''
-              require("lsp-format").on_attach(client)
-            '';
-        })
-      );
+    autoGroups.nixvim_lsp_fmt_setup.clear = false;
 
-    };
+    autoCmd = lib.optionals (cfg.lspServersToEnable != "none") [
```

Comment:
Was literally just typing a suggestion to move the condition here! :exploding_head:

(nit: I usually prefer to use `mkIf` as it actually avoids making any definitions, using `optionals` will always _define_ `autoCmd = [ ]`, potentially overriding lower-prio definitions)

Reply:
Ah, good to know! I'll keep that in mind for the future.